doc/tutorial/ApiType.lhs

```diff
@@ -321,11 +321,32 @@ data BasicAuth (realm :: Symbol) (userData :: *)
 Which is used like so:
 
 ``` haskell
-type ProtectedAPI12
+type ProtectedAPI11
      = UserAPI                              -- this is public
  :<|> BasicAuth "my-realm" User :> UserAPI2 -- this is protected by auth
 ```
 
+### Empty APIs
+
+Sometimes it is useful to be able to generalise an API over the type of some
+part of it:
+
+``` haskell
+type UserAPI12 innerAPI
+     = UserAPI             -- this is the fixed bit of the API
+ :<|> "inner" :> innerAPI  -- this lets us put various other APIs under /inner
+```
+
+If there is a case where you do not have anything extra to serve, you can use
+the `EmptyAPI` combinator to indicate this:
+
+``` haskell
+type UserAPI12Alone = UserAPI12 EmptyAPI
+```
+
+This also works well as a placeholder for unfinished parts of an API while it
+is under development.
```

Comment:
To give a concrete idea, it might be nice to mention that this is somewhat equivalent to how we use `undefined` (yeah, bad bad bad, I know =P) or `()` or what have you when we need to fill in an expression but don't want to worry about the actual value of it until later.